pkg/cli/root.go

```diff
@@ -423,14 +423,21 @@ func isDefaultPolicyConfig(cfg *config.Config) bool {
 		for _, policy := range repository.Policies {
 			if len(policy.Actions)+len(policy.Users) > 0 {
 				log.Info().Interface("repository", repository).
-					Msg("repository with non-empty policy detected, default authorization disabled")
+					Msg("repository with non-empty policy detected, anonymous authorization disabled")
+
+				return false
+			}
+
+			if len(repository.DefaultPolicy) > 0 {
+				log.Info().Interface("repository", repository).
+					Msg("repository with default policy detected, anonymous authorization disabled")
 
 				return false
 			}
 		}
 	}
 
-	log.Info().Msg("default authorization detected")
+	log.Info().Msg("anonymous authorization detected")
```

Comment:
So where do we check if anonymous policies are actually present in the config? So far we only check if other types of policies are missing.

This was probably not needed before as the read access flag was used in some other place.

Reply:
config needs to miss all default+admin+users policies right?
if config misses all of them it means we have at least one anonymous policy, otherwise config.HTTP.RawAccessControl would be nil and the check would not run.

maybe I'm missing something :D